### PR TITLE
Make MirageJS peer dependency

### PR DESCRIPTION
**Pull request by cah-brian-gantzler**
This makes `MirageJS` a peer dependency. This is a breaking change. Any existing applications using `ember-cli-mirage` will have to rerun the default blueprint `ember install ember-cli-mirage` or add `miragejs` to your `package.json` mannually.

Fixes: https://github.com/miragejs/ember-cli-mirage/issues/2416

### CODE REVIEW

File: packages/ember-cli-mirage/blueprints/ember-cli-mirage/index.js [hunk 1]
```diff
@@ -30,17 +29,9 @@ module.exports = {
...
-      var shutdownText =
-        '  if (window.server) {\n    window.server.shutdown();\n  }';
-      return this.insertIntoFile('tests/helpers/destroy-app.js', shutdownText, {
-        after: "run(application, 'destroy');\n",
-      });
-    }
-  },
-
   afterInstall: function () {
-    return this.insertShutdownIntoDestroyApp();
```
Comment (cah-briangantzler) [inline packages/ember-cli-mirage/blueprints/ember-cli-mirage/index.js:44]: This was removed as `destoy-app.js` was removed in `3.0.0`

File: packages/ember-cli-mirage/package.json [hunk 2]
```diff
@@ -101,7 +100,8 @@
   "peerDependencies": {
     "@ember/test-helpers": "*",
     "ember-data": "*",
-    "ember-qunit": "*"
+    "ember-qunit": "*",
```
Comment (cah-briangantzler) [inline packages/ember-cli-mirage/package.json:103]: Should these be `*`?

**Comment by cah-briangantzler** [comment]
Would you be opposed to adding [Husky](https://github.com/typicode/husky). I have it on my own apps so keep forgetting to run lint and prettier prior to submitting PRs :)

**Comment by ef4** [comment]
This is a good change, without it people cannot reliably adopt the new pattern that is documented for 3.0 (importing directly from miragejs).

**Review by SergeAstapov — APPROVED** [review]

**Pull request merged**